### ISSUE TITLE
Change in number of segments on a different run

### ISSUE THREAD

**Issue by Ankithas93** [thelastpickle/cassandra-reaper#673]
**Spotted on 1.4.0 (4c225e6)**

I recently upgraded my cassandra clusters from 2.1.16 to 3.11.4. 
Cluster 1 and Cluster 2 - 24 nodes with num_tokens set to 256(Vnodes)
Before the upgrade the segments to be repaired on the nodes were 6144 (24 nodes* 256 Vnodes). After upgrade I tried to start repair.Cluster1's total segment count to be repaired is 536 and cluster2's  total segment count to be repaired is 1284.
I did not change anything except upgrading it. How can I know that repair is running on all data of my cluster? Or is there some change with respect to the cassandra version?

**Comment by adejanovski** [comment]
Ain't it awesome? 😄 

Reaper uses improvements from Cassandra 3.x to drastically reduce the number of segments. What happens is that in Cassandra 3.0/3.x, if we trigger a repair with several token ranges which have the exact same replicas, they will be processed in a single session.
Reaper then tries to reduce the number of segment by grouping all such token ranges together, still trying to get a good balance of tokens across segments.
Apparently, cluster2's token distribution gives a less efficient reduction, which could be due to the randomness of token allocation on bootstrap (or the number of segments per node that is defined in the repair).

To sum it up : it's expected and even desired that you can handle repairs with less segments in Cassandra 3.0 and above. All of your data is still repaired.

Repair should be faster now. If some segments start to take too long, tune the number of segments per node.

Cheers
